modules/library/referencing/src/test/java/org/geotools/referencing/EnvelopeReprojectorTest.java

```diff
@@ -0,0 +1,247 @@
+
+    public static final double EPS = 0.01;
+
+    /** Tests the transformations of an envelope. */
+    @Test
+    // code is using equals with extra parameters and semantics compared to the built-in equals
+    @SuppressWarnings("PMD.SimplifiableTestAssertion")
+    public void testEnvelopeTransformation() throws FactoryException, TransformException {
+        final CoordinateReferenceSystem mapCRS = CRS.parseWKT(WKT.UTM_10N);
+        final CoordinateReferenceSystem WGS84 = DefaultGeographicCRS.WGS84;
+        final CoordinateOperation operation =
+                CRS.getCoordinateOperationFactory(true).createOperation(WGS84, mapCRS);
+        assertFalse(operation.getMathTransform().isIdentity());
+
+        final GeneralBounds firstEnvelope =
+                new GeneralBounds(new double[] {-124, 42}, new double[] {-122, 43});
+        firstEnvelope.setCoordinateReferenceSystem(WGS84);
+
+        final GeneralBounds transformedEnvelope =
+                EnvelopeReprojector.transform(operation, firstEnvelope);
+        transformedEnvelope.setCoordinateReferenceSystem(mapCRS);
+
+        final CoordinateOperation inverse =
+                CRS.getCoordinateOperationFactory(true).createOperation(mapCRS, WGS84);
+        final GeneralBounds oldEnvelope =
+                EnvelopeReprojector.transform(inverse, transformedEnvelope);
+        oldEnvelope.setCoordinateReferenceSystem(WGS84);
+
+        assertTrue(oldEnvelope.contains(firstEnvelope, true));
+        assertTrue(oldEnvelope.equals(firstEnvelope, 0.02, true));
+    }
+
+    /**
+     * Tests the transformations of an envelope when the two CRS have identify transforms but
+     * different datum names
+     */
+    @Test
+    public void testEnvelopeTransformation2() throws FactoryException, TransformException {
+        final CoordinateReferenceSystem WGS84Altered = CRS.parseWKT(WKT.WGS84_ALTERED);
+        final CoordinateReferenceSystem WGS84 = DefaultGeographicCRS.WGS84;
+        final MathTransform crsTransform = CRS.findMathTransform(WGS84, WGS84Altered, true);
+        assertTrue(crsTransform.isIdentity());
+
+        final GeneralBounds firstEnvelope =
+                new GeneralBounds(new double[] {-124, 42}, new double[] {-122, 43});
+        firstEnvelope.setCoordinateReferenceSystem(WGS84);
+
+        // this triggered a assertion error in GEOT-2934
+        Bounds transformed = CRS.transform(firstEnvelope, WGS84Altered);
+
+        // check the envelope is what we expect
+        assertEquals(transformed.getCoordinateReferenceSystem(), WGS84Altered);
+        double EPS = 1e-9;
+        assertEquals(transformed.getMinimum(0), firstEnvelope.getMinimum(0), EPS);
+        assertEquals(transformed.getMinimum(1), firstEnvelope.getMinimum(1), EPS);
+        assertEquals(transformed.getMaximum(0), firstEnvelope.getMaximum(0), EPS);
+        assertEquals(transformed.getMaximum(1), firstEnvelope.getMaximum(1), EPS);
+    }
+
+    @Test
+    @Ignore
```

Comment:
Is it intentionally ignored?
Ah, this class is a rename of the previous one and that was already ignored, there... So I think it's fine as it is

Reply:
Yes indeed, I moved the tests that work 99% on the EnvelopeReprojector to their dedicated test class.